genie/validate.py

```diff
@@ -415,3 +416,61 @@ def standardize_string_for_validation(
         return standardized_str
     else:
         return input_string
+
+
+def get_invalid_allele_rows(
+    input_data: pd.DataFrame,
+    input_col: str,
+    allowed_alleles: list,
+    ignore_case: bool = False,
+) -> pd.Index:
+    """
+    Find invalid indices in a DataFrame column based on allowed allele values.
+
+    Args:
+        input_data (pd.DataFrame): The DataFrame to search.
+        input_col (str): The name of the column to check.
+        allowed_alleles (list): The list of allowed allele values.
+        ignore_case (bool, optional): whether to perform case-insensitive matching
+
+    Returns:
+        pd.Index: A pandas index object indicating the row indices that
+        don't match the allowed alleles
+    """
+    search_str = rf"^[{''.join(allowed_alleles)}]+$"
```

Comment:
I think this regex does actually check if for alleles like 'ATATATATAT'

Reply:
It does. Have to update it for '-', empty strings and NAs based on our discussion